libkmod/libkmod-index.c

```diff
@@ -25,10 +25,9 @@
  * Integers are stored as 32 bit unsigned in "network" order, i.e. MSB first.
  * All files start with a magic number.
  *
- * Magic spells "BOOTFAST". Second one used on newer versioned binary files.
- * #define INDEX_MAGIC_OLD 0xB007FA57
+ * Magic spells "BOOTFAST". Deprecated versions encoded it as 0xB007FA57.
```

Comment:
Which versions are deprecated - where those produced with kmod v1, v2 or even earlier with module-init-tools?

Reply:
The referenced file had this old magic number even commented out in version 1. Definitely deprecated. Since I didn't start to investigate module-init-tools to trace its origin, it's save to say that whatever software version used it, is deprecated.

Reply:
/me reporting for my archaeology class
 
```
commit 44d7ac4f9f61db23a8d9feb74d31a0e97df7c6a5
Author: Jon Masters <jcm@jonmasters.org>
Date:   Fri Jan 23 01:05:48 2009 -0500

    add versioning to the binary module files
    
    We need to have a version attached to the binary trie in case we ever
    need to change the format in some fashion. So this patch does that, and
    also changes the MODULE_MAGIC to 0xB007F457 (change the 'A' to a '4')
    so that we can notice that we have the second version of the files.
    
    Signed-off-by: Jon Masters <jcm@jonmasters.org>
```

a few weeks later, the old version got commented out.

Reply:
Wait a minute ... TIL there is a git repo for module-init-tools :sweat_smile: 

Would be amazing if we something like the ^^ details as an inline comment, the cryptic one currently.

Reply:
I think we had a repo for modutils too (that got replaced by module-init-tools in the 2.4 -> 2.6 transition, but all I can find now are the tarballs and binary packages: https://www.kernel.org/pub/linux/utils/kernel/modutils/